src/magic_folder/test/test_magic_file.py

```diff
@@ -324,10 +326,136 @@ def test_multiple_remote_updates(self):
         mf = self.magic_file_factory.magic_file_for(abspath)
-        self.participants.add(create_author("beth", VerifyKey(b"\xff" * 32)), random_dircap())
-        self.participants.add(create_author("callum", VerifyKey(b"\xee" * 32)), random_dircap())
-        self.participants.add(create_author("dawn", VerifyKey(b"\xee" * 32)), random_dircap())
+        self.participants.add("beth", random_dircap())
+        self.participants.add("callum", random_dircap())
+        self.participants.add("dawn", random_dircap())
         d0 = mf.found_new_remote(remote0, self.participants.participants[1])
         d1 = mf.found_new_remote(remote0, self.participants.participants[2])
         d2 = mf.found_new_remote(remote0, self.participants.participants[3])
         yield DeferredList([d0, d1, d2])
+
+    @inlineCallbacks
+    def test_not_conflicted(self):
+        """
+        """
+        relpath = "not-conflict"
+        abspath = self.config.magic_path.preauthChild(relpath)
+        mf = self.magic_file_factory.magic_file_for(abspath)
+        with self.assertRaises(ResolutionError):
+            yield mf.resolve_conflict("foo")
+
+    @inlineCallbacks
+    def test_remote_conflict(self):
+        """
+        We have a conflict
+        """
+        relpath = "dual-conflict"
+        cap0 = random_immutable(directory=True)
+        cap1 = random_immutable(directory=True)
+        remote0 = RemoteSnapshot(
+            relpath=relpath,
+            author=create_author("someone", VerifyKey(b"\xff" * 32)),
+            metadata={"modification_time": 0},
+            capability=cap0,
+            parents_raw=[],
+            content_cap=random_immutable(),
+            metadata_cap=random_immutable(),
+        )
+        remote1 = RemoteSnapshot(
+            relpath=relpath,
+            author=create_author("someone_else", VerifyKey(b"\xff" * 32)),
+            metadata={"modification_time": 0},
+            capability=cap1,
+            parents_raw=[],
+            content_cap=random_immutable(),
+            metadata_cap=random_immutable(),
+        )
+        self.remote_cache._cached_snapshots[cap0.danger_real_capability_string()] = remote0
+        self.remote_cache._cached_snapshots[cap1.danger_real_capability_string()] = remote1
+
+        abspath = self.config.magic_path.preauthChild(relpath)
+        mf = self.magic_file_factory.magic_file_for(abspath)
+        self.participants.add("beth", random_dircap())
+        d0 = mf.found_new_remote(remote0, self.participants.participants[0])
+        d1 = mf.found_new_remote(remote1, self.participants.participants[1])
+        yield DeferredList([d0, d1])
+
+        self.assertEquals(
+            self.config.list_conflicts_for("dual-conflict"),
+            [Conflict(snapshot_cap=cap1, participant_name="beth")]
+        )
+
+        self.uploader._uploads = ["dual-conflict"]
+        snap = yield mf.resolve_conflict(Conflict(snapshot_cap=cap1, participant_name="beth"))
+        print("XXX", snap)
```

Comment:
Should this print statement be removed? If it's intended to be preserved, consider using more descriptive string here.